index.bs

```diff
@@ -1033,23 +1031,29 @@ dictionary CompilationOptions {
 };
 
 interface Model {
-  Promise<Compilation> createCompilation(optional CompilationOptions options = {});
+  Promise<Compilation> compile(optional CompilationOptions options = {});
```

Comment:
With this API, the computation is tied with the compilation. should the compile method have the inputs/outputs pair and execution can only execution the graph compiled with the input/output pair? Otherwise, the compilation might not support the execution of the sub-graph scenario?

Reply:
@pyu10055 Good idea. Will this work now?

Reply:
Do we need this change?

@pyu10055 , according to https://github.com/webmachinelearning/webnn/issues/87#issuecomment-694330940, you seemed to agree the ` Promise<NamedOutputs> compute(NamedInputs inputs, optional NamedOutputs outputs);` (the ability to specify the outputs when executing) would support the sub-graph execution use case, correct?

Reply:
@huningxin I actually do not recommend partial execution of a compiled graph as any operator in between the input and output operands may be folded with the surrounding operators from fusions or even eliminated by constant folding. The only guaranteed executable operands are the input and the output operands of the compiled graph. Partial compilation of a model is much more robust. Once a compilation is produced for the partial model, it stays valid and can be executed as such.

Reply:
@wchao1115 , I only mean the `compute` method now allows developers to select the output operands of a compiled graph. I didn't recommend the partial execution for any intermediate operands between inputs and outputs.

My point is that the graph building is in the context of new `ModelBuilder` object. Developers are free to create a sub-graph by `createModel` API and compile/execute it. I think it would support the sub-graph execution use case without adding inputs/outputs pair to `compile` method.

Reply:
@huningxin and I discussed this specific change offline. We want to track partial graph execution as a separate issue. @pyu10055 you may create a new issue just for this topic if you wish. I will now complete this PR with the original set of changes as outlined in the PR description above.

Reply:
Created https://github.com/webmachinelearning/webnn/issues/105 as follow-up.